File: DAY3/problem_3.py
# def reverseArray(arr):
#     left=0
#     right=len(arr)-1

#     while left < right:
#         arr[left],arr[right]=arr[right],arr[left]

#         left+=1
#         right-=1

# if __name__=="__main__":
#     arr=[1,4,3,2,6,5]
#     reverseArray(arr)

#     for i in range (len(arr)):
#         print(arr[i],end=" ")



# Approch 2
def reversearray(arr):
    n=len(arr)

    for i in range(n//2):
        temp=arr[i]
        arr[i]=arr[n-1-i]
        arr[n-1-i]=temp          

if __name__=="__main__":
    arr=[1,4,3,2,6,5]
    reversearray(arr)

    for i in range (len(arr)):
        print(arr[i],end=" ")

File: DAY3/problem_3.java
//Approach 1
// public class problem_3 {

//     public void reverseArray(int arr[]){
//         int left=0;
//         int right=arr.length-1;

//         while(left<right){
//             int temp;
//             temp=arr[left];
//             arr[left]=arr[right];
//             arr[right]=temp;

//             left++;
//             right--;

//         }
//     }
//     public static void main(String[] args) {
//         problem_3 obj=new problem_3();
//         int arr[]={1,4,3,2,6,5};
//         obj.reverseArray(arr);
//         for (int num:arr){
//             System.out.print(num+" ");
//         }
//     }

// }

//Approach 2

public class problem_3{

    public void reverseArray(int arr[]){
                int n=arr.length;
        
               for (int i=0;i<n/2;i++){
                 //swap
                 int temp=arr[i];
                 arr[i]=arr[n-i-1];
                 arr[n-i-1]=temp;
               }
            }
public static void main(String[] args) {
        problem_3 obj=new problem_3();
        int arr[]={1,4,3,2,6,5};
        obj.reverseArray(arr);

        for(int num:arr){
            System.out.print(num+" ");
        }
    }

}